src/ControlSystem/ControlErrors/SizeControlInfo.cpp

```diff
@@ -0,0 +1,22 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "ControlSystem/ControlErrors/SizeControlInfo.hpp"
+
+CrossingTimeInfo::CrossingTimeInfo(
+    const double char_speed_crossing_time,
+    const double comoving_char_speed_crossing_time,
+    const double delta_radius_crossing_time)
+    : t_char_speed(char_speed_crossing_time),
+      t_comoving_char_speed(comoving_char_speed_crossing_time),
+      t_delta_radius(delta_radius_crossing_time) {
+  if (t_char_speed > 0.0) {
+    if (t_delta_radius > 0.0 and t_delta_radius <= t_char_speed) {
```

Comment:
[Optional] This sort of thing would be simpler if you didn't use 0 to mean infinity.  The same pattern occurs throughout this PR.

Reply:
I think I want to wait to see what the final ZeroCrossingPredictor looks like (because t_char_speed etc will come from the ZeroCrossingPredictor) before changing this from zero to something else.  But yes, instead of zero I am open to using a `std::optional` or maybe `std:::numeric_limits<double>::infinity` to denote "doesn't cross zero or cannot determine when it crosses zero".